app/models/role_loader.rb

```diff
@@ -0,0 +1,45 @@
+class RoleLoader
+
+  def available_roles_for_application(name)
+    load_organisation_types.map do |organisation_type, data|
+      data["available_roles"].map do |role_name, applications|
+        Role.new role_name, applications if applications.include?(name)
+      end
+    end.flatten.compact.uniq.sort
```

Comment:
Wouldn't it be better to `uniq` on the available roles before creating a bunch or Role objects?